script.js

```diff
@@ -1,4 +1,159 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
-};
+var customerName = prompt("Welcome to a game of Scissors, Paper, Stone! Please enter your name", "You name goes here!");
+var gameMode = prompt(
+  "Welcome to a game of Scissors, Paper, Stone! Please enter your game mode, i.e og, reversed, korean"
+);
+
+// total number of wins for player, computer
+var userScoreCount = 0;
+var comScoreCount = 0;
+var winner = '';
+
+// SPS
+var computerRandom = function() {
+  var ranNum = Math.floor(Math.random() * 3);
+  console.log(ranNum)
+  if (ranNum == 0) {
+    return 'stone';
+  } else if(ranNum == 1) {
+    return 'paper' ;
+  }
+  return 'scissors';
+}
+
+var main = function(input) {
+  var userName = customerName;
+  var computerChoice = computerRandom(); // calling a function
+  
+  if (gameMode == "og") {
+    return og(input)
+  }
+  if (gameMode == "reversed") {
+    return reversed(input)
+  }
+  if (gameMode == "korean") {
+    return korean(input)
+  } 
+  return 'Invalid response. Please only type "og", "reversed", "korean"!'
+}
+
+
+
+var og =function(input){
+  var userName = customerName;
+  var computerChoice = computerRandom(); // calling a function
+  if (
+    input != 'scissors' && input != 'paper' && input != 'stone'
+  ){
+    return 'There are only 3 input options "scissors" "paper" or "stone". Please try again!'
+  }
+  else if (
+    input == computerChoice
+  ){
+    userScoreCount = userScoreCount;
+    comScoreCount = comScoreCount;
+    return ` Hi ${userName}, you have typed ${input}. <br> Computer played ${computerChoice}. Your current score is at ${userScoreCount} and computer's current score is at ${comScoreCount}. <br> It is a tie!`
+  }
+  else if (
+    input == 'scissors' && computerChoice == 'paper' ||
+    input == 'paper' && computerChoice == 'stone' ||
+    input == 'stone' && computerChoice == 'scissors')
+  {
+    userScoreCount = userScoreCount +1;
+    comScoreCount = comScoreCount;
+    return `Congratulations ${userName}, you win! <br> You typed ${input}. <br> Computer played ${computerChoice}. <br> Your current score count is at ${userScoreCount} and computer's score count is at ${comScoreCount}.`
+  }
+  else if (
+    input == 'scissors' && computerChoice == 'stone' ||
+    input == 'paper' && computerChoice == 'scissors' || 
+    input == 'stone' && computerChoice == 'paper') 
+      comScoreCount = comScoreCount + 1;
+      userScoreCount = userScoreCount;
```

Comment:
this could be simplified as an else, where the conditions do not have to be specified as the win and draws have already been covered. can save on typing some code.